.github/actions/early-terminator/action.yml

```diff
@@ -0,0 +1,39 @@
+name: "early terminate"
+description: |
+  Terminate CI workflow earlier once job failure is detected
+inputs:
+  github-token:
+    description: 'Github token'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Post test results on PR
+      uses: actions/github-script@v4.0.2
+      with:
+        script: |
+            // use github REST apis: https://docs.github.com/en/rest/reference/actions#workflow-runs
+            const https = require('https');
+            const options = {
+              hostname: 'api.github.com',
+              path: `/repos/${process.env.GITHUB_REPOSITORY}/actions/runs/${process.env.GITHUB_RUN_ID}/cancel`,
+              headers: {
+                'Authorization': `token ${{ inputs.github-token }}`,
```

Comment:
Given this uses the github token, do we know how this is supposed to behave on Ci run on PRs?

Reply:
I have done canary run with passing in GH token, https://github.com/diem/diem/actions/runs/1401703623

Reply:
No i'm specifically asking about what happens when Ci runs on a PR, not being run as apart of a land or canary step on `auto` or `canary` branch

Reply:
this early termination is only added in ci-workflow which I assume it is only triggered by auto and canary branches? If i misunderstood please correct me

Reply:
No ci-test workflow is triggered on PRs as well as on auto and canary branches

Reply:
Since we have other actions are also using GHA token as passing in value, I think it should be ok but we can do some tests to confirm. 
```
id: changes
name: determine changes
uses: diem/actions/changes@faadd16607b77dfa2231a8f366883e01717b3225
with:
    workflow-file: ci-test.yml
    github-token: ${{secrets.GITHUB_TOKEN}}

LOGIN=( "--header" "authorization: Bearer ${GITHUB_TOKEN}" )
```